packages/phase2cli/src/lib/utils.ts

```diff
@@ -29,22 +27,47 @@ import {
 } from "@zkmpc/actions"
 import { fileURLToPath } from "url"
 import path from "path"
+import { GithubAuthProvider, OAuthCredential } from "firebase/auth"
 import {
     FirebaseDocumentInfo,
-    FirebaseServices,
     ParticipantContributionStep,
     ParticipantStatus,
     Timing,
     VerifyContributionComputation
 } from "../../types/index"
 import { collections, emojis, numIterationsExp, paths, symbols, theme } from "./constants"
-import { initServices } from "./firebase"
 import { GENERIC_ERRORS, GITHUB_ERRORS, showError } from "./errors"
 import { downloadLocalFileFromBucket } from "./storage"
 import { getAllCeremonies } from "./queries"
 
 dotenv.config()
 
+/**
+ * Exchange the Github token for OAuth credential.
+ * @param githubToken <string> - the Github token generated through the Device Flow process.
+ * @returns <OAuthCredential>
+ */
+export const exchangeGithubTokenForCredentials = (githubToken: string): OAuthCredential =>
+    GithubAuthProvider.credential(githubToken)
+
+/**
+ * Get the Github handle associated to the account from which the token has been generated.
+ * @param githubToken <string> - the Github token.
+ * @returns <Promise<any>> - the Github handle of the user.
+ */
+export const getGithubUserHandle = async (githubToken: string): Promise<any> => {
+    // Ask for user account public information through Github API.
+    const response = await request("GET https://api.github.com/user", {
+        headers: {
+            authorization: `token ${githubToken}`
+        }
+    })
+
+    if (response) return response.data.login
```

Comment:
might be worth checking the status code of the response before returning?

Reply:
Correct! Thanks for the heads up, will fix in next commit.

Reply:
Fixed [bf0fbab](https://github.com/quadratic-funding/mpc-phase2-suite/pull/270/commits/bf0fbaba3ab89b6d57dcd4dfa410129e55e04a2c)